### PR TITLE
Visual Studio Code - Project Manager: Fixes `undefined` property access when checking project tags.

### LINKED ISSUE #264: [Extension Bug] Visual Studio Code - Project Manager
Raycast version: 1.25.2

## Description

[Visual Studio Code - Project Manager](https://www.raycast.com/MarkusLanger/vscode-project-manager) does not work if you do not have tags in any project.
For some reason my `projects.json` had no `tags` key, which the extension directly checks, being undefined it breaks the extension

## Steps To Reproduce

1. Install Extension
2. Remove any `tags` key from your `projects.json`
3. Run `Search Project Manager` command

## The current behavior
Shows `Something went wrong`

## The expected behavior
List projects from the Project Manager extension

## Log (deobfuscated)
```ts
TypeError: Cannot read properties of undefined (reading 'length')

search-project-manager-projects.js:1:1331

---
function O(t) {
    let o = new Map;
    return t.forEach(r => {
        (r.tags.length > 0 ? r.tags : ["[no tags]"]).forEach(i => {
/-----------------^
            let c = [];
            o.has(i) && c.push(...o.get(i) || []), c.push(r), o.set(i, c)
        })
    }), new Map([...o.entries()].sort())
}
---

    at Array.forEach (<anonymous>)
O:search-project-manager-projects.js:1:1311
b:search-project-manager-projects.js:1:1495
g:search-project-manager-projects.js:3:66
Sl:react-reconciler.production.min.js:5:19049
Wi:react-reconciler.production.min.js:9:6948
Fi:react-reconciler.production.min.js:9:939
Jo:react-reconciler.production.min.js:9:870
Ut:react-reconciler.production.min.js:9:727
```

### PR DESCRIPTION
**Pull request by rybertm**
Closes #264

## Description

Fixed a bug when accessing project tags

## Type of change

- Bug fix / improvement for somebody else's extension. Extension author: @GH-handle

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I checked that extension's icon has size of `512x512`

### CODE REVIEW

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @rybertm 👋 

I couldn't reproduce this error in #264 but I think if we can catch anything that could potentially break, let's handle it 💯 

I tested with an object containing no tag property, one with a tag and one with no tags.

If I manually remove the tags line from a project in the config file, then there is an error.

XOT0ChJCHVsxXomb

Maybe add something like ```accessoryTitle={tags?.join(", ")}``` to handle this scenario.

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by rybertm** [comment]
Interesting, it happens to me because my project config file did not have a tag property at all.

> If I manually remove the tags line from a project in the config file, then there is an error.

Yeah, I didn't check for any other usage of the `tag` property and didn't debug it after my changes, so I missed this, sorry.

> Maybe add something like accessoryTitle={tags?.join(", ")} to handle this scenario.

Done.

File: extensions/visual-studio-code-project-manager/src/search-project-manager-projects.tsx [hunk 1]
```diff
@@ -38,7 +38,7 @@ function getProjectsGroupedByTag(projects: ProjectEntry[]): Map<string, ProjectE
   const groupedProjects = new Map<string, ProjectEntry[]>();
 
   projects.forEach((project: ProjectEntry) => {
-    const tags = project.tags.length > 0 ? project.tags : ["[no tags]"];
+    const tags = project.tags?.length ?? 0 > 0 ? project.tags : ["[no tags]"];
```
Comment (pernielsentikaer) [inline extensions/visual-studio-code-project-manager/src/search-project-manager-projects.tsx:41]:
```suggestion
    const tags = project.tags?.length > 0 ? project.tags : ["[no tags]"];
```

This should be sufficient in this case

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
This is almost ready to get shipped 🚢 

Could you please correct lines 60-61, a project can potentially be added with multiple tags giving the error: ```Warning: Encountered two children with the same key``` to this:

```
        {value?.map((project, index) => (
          <ProjectListItem key={project.rootPath+index} {...project} />
```

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Review by pernielsentikaer — APPROVED** [review]
Should be good to be merged :rocket:

**Comment by pernielsentikaer** [comment]
Are you also okay with this @MarkusLanger

**Comment by rybertm** [comment]
Sorry, I'm not used to PR's, so I updated my main, which was the branch I made the changes, with upstream, so I just reset it back to the last commit I made. If there is anything wrong or if I need to change anything just let me know.

Again, sorry about that @pernielsentikaer.

**Comment by MarkusLanger** [comment]
See also PR #258

**Comment by pernielsentikaer** [comment]
@rybertm We have enabled package.json linter, could you please look into the error and correct it (preferences now needs a label)

**Pull request merged**